tests/torch/quantization/test_strip.py

```diff
@@ -409,3 +414,140 @@ def test_nncf_strip_lora_model(mode, decompressor_class, torch_dtype, atol, mock
             expected_class=decompressor_class,
         )
         assert torch.allclose(compressed_output, stripped_output, atol=atol)
+
+
+SIGNED_WEIGHT_SAMPLE = [-1.0, -0.75, -0.5, -0.25, 0.0, 0.25, 0.5, 0.75]
+SCALE_SAMPLE = [2.0]
+
+
+@pytest.mark.parametrize(
+    ("num_bits", "ref_scale", "torch_dtype"),
+    (
+        (4, [0.25], torch.float32),
+        (8, [0.01563], torch.float32),
+        (4, [0.25], torch.float16),
+        (8, [0.01563], torch.float16),
+        (4, [0.25], torch.bfloat16),
+        (8, [0.01563], torch.bfloat16),
+    ),
+)
+def test_sym_fq_to_decompressor(num_bits, ref_scale, torch_dtype):
+    weights_shape = (1, len(SIGNED_WEIGHT_SAMPLE))
+    weight = torch.tensor(SIGNED_WEIGHT_SAMPLE)
+    weight = weight.expand(weights_shape).to(torch_dtype)
+
+    scale_shape = (1, 1)
+    scale = torch.tensor(SCALE_SAMPLE)
+    scale = scale.expand(scale_shape).to(torch.float16)
+
+    # reference scale calculates with this formula:
+    # levels = (2 ** num_bits)
+    # level_low = -(levels // 2)
+    # ref_scale = SCALE_SAMPLE / abs(level_low)
+    ref_scale = torch.tensor(ref_scale)
+    ref_scale = ref_scale.expand(scale_shape).to(torch.float16)
+
+    qspec = PTQuantizerSpec(
+        num_bits=num_bits,
+        mode=QuantizationMode.SYMMETRIC,
+        signedness_to_force=True,
+        narrow_range=False,
+        scale_shape=scale.shape,
+        logarithm_scale=False,
+        half_range=False,
+        is_quantized_on_export=True,
+    )
+    lspec = PTLoraSpec(
+        lora_rank=1,
+        orig_weight_shape=weight.shape,
+        weight_shape=weight.shape,
+    )
+
+    quantizer = SymmetricLoraQuantizer(qspec, lspec)
+    quantizer.scale.data = scale
+
+    with torch.no_grad():
+        decompressor, q_weight = sym_fq_to_decompressor(
+            quantizer,
+            weight,
+        )
+
+    qdq_weight = (q_weight * ref_scale).to(torch_dtype)
+
+    assert torch.allclose(qdq_weight, weight)
+    assert torch.allclose(decompressor._scale, ref_scale)
+
+
+UNSIGNED_WEIGHT_SAMPLE = [0.0, 0.1, 0.2, 0.3, 0.4, 0.5]
+INPUT_LOW_SAMPLE = [0.0]
+INPUT_RANGE_SAMPLE = [0.5]
+
+
+@pytest.mark.parametrize(
+    ("num_bits", "ref_scale", "ref_zero_point", "torch_dtype"),
+    (
+        (4, 0.03333, 0.0, torch.float32),
+        (8, 0.00196, 0.0, torch.float32),
+        (4, 0.03333, 0.0, torch.float16),
+        (8, 0.00196, 0.0, torch.float16),
+        (4, 0.03345, 0.0, torch.bfloat16),
+        (8, 0.007812, 0.0, torch.bfloat16),
```

Comment:
It's so strange that the reference scales for 8-bit are much different from those for 4-bit.

Reply:
This is normal. For the provided numbers, in bfloat16, the scale is calculated lower than bf16.eps, so it was replaced.